taipy/core/data/couchbase_collection_datanode.py

```
@@ -0,0 +1,183 @@
+        __DB_NAME_KEY,
+        __COLLECTION_KEY,
+    ]
+
+    def __init__(
+        self,
+        config_id: str,
+        scope: Scope,
+        id: Optional[DataNodeId] = None,
+        owner_id: Optional[str] = None,
+        parent_ids: Optional[Set[str]] = None,
+        last_edit_date: Optional[datetime] = None,
+        edits: List[Edit] = None,
+        version: str = None,
+        validity_period: Optional[timedelta] = None,
+        edit_in_progress: bool = False,
+        editor_id: Optional[str] = None,
+        editor_expiration_date: Optional[datetime] = None,
+        properties: Dict = None,
+    ) -> None:
+        if properties is None:
+            properties = {}
+        required = self._REQUIRED_PROPERTIES
+        if missing := set(required) - set(properties.keys()):
+            raise MissingRequiredProperty(
+                f"The following properties {', '.join(missing)} were not informed and are required."
+            )
+
+        super().__init__(
+            config_id,
+            scope,
+            id,
+            owner_id,
+            parent_ids,
+            last_edit_date,
+            edits,
+            version or _VersionManagerFactory._build_manager()._get_latest_version(),
+            validity_period,
+            edit_in_progress,
+            editor_id,
+            editor_expiration_date,
+            **properties,
+        )
+
+
+         # Create a Couchbase connection using the provided properties.
+         # For more information on connecting to Couchbase, see:
+    # https://docs.couchbase.com/python-sdk/current/hello-world/start-using-sdk.html
+        try:
+            self.cluster = Cluster(
+                f"couchbase://{properties.get(self.__DB_HOST_KEY, self.__DB_HOST_DEFAULT)}",
+                ClusterOptions(
+                    PsswordAuthenticator(
+                        properties.get(self.__DB_USERNAME_KEY, ""),
+                        properties.get(self.__DB_PASSWORD_KEY, "")
```

I'm quite sure that Couchbase requires username and password to connect.

If the `properties` doesn't contain the `username` and `password`, we should raise an error before this, meaning that we should add the `self.__DB_USERNAME_KEY` and `self.__DB_PASSWORD_KEY` to the `self._REQUIRED_PROPERTIES`